src/whitenoise/compress.py

```diff
@@ -175,12 +179,14 @@ def main(argv=None):
         use_brotli=args.use_brotli,
         quiet=args.quiet,
     )
-    for dirpath, _dirs, files in os.walk(args.root):
-        for filename in files:
-            if compressor.should_compress(filename):
-                path = os.path.join(dirpath, filename)
-                for _compressed in compressor.compress(path):
-                    pass
+
+    with ThreadPoolExecutor() as executor:
+        for dirpath, _dirs, files in os.walk(args.root):
+            for filename in files:
+                if compressor.should_compress(filename):
+                    executor.submit(
```

Comment:
If compressor.compress raises you lose the exception

Reply:
#615